src/Mod/TechDraw/App/DrawUtil.cpp

```diff
@@ -1887,6 +1887,14 @@ std::string DrawUtil::cleanFilespecBackslash(const std::string& filespec)
     return noBackslash;
 }
 
+//! returns true if the Gui module and its event loop are active.
+bool DrawUtil::isGuiUp()
+{
+    std::string pyCommand{"import FreeCAD\nguiState = FreeCAD.GuiUp"};
+    std::string result{Base::Interpreter().runStringWithKey(pyCommand.c_str(), "guiState", "None")};
+    return  (result == "1");
+}
```

Comment:
MergeDocument has a [c++ implementation](https://github.com/FreeCAD/FreeCAD/blob/main/src/App/MergeDocuments.cpp#L91-L94) of this, here it would look like:
```suggestion
bool DrawUtil::isGuiUp()
{
    auto* app = QCoreApplication::instance();
    return app && app->inherits("QApplication");
}
```
Maybe it fits your needs better than to go through the interpreter and back?